components/nimbus/src/enrollment.rs

```diff
@@ -1182,9 +1205,11 @@ pub enum EnrollmentChangeEventType {
     EnrollFailed,
     Disqualification,
     Unenrollment,
+    #[allow(unused)]
     UnenrollFailed,
 }
 
+#[cfg(feature = "stateful")]
```

Comment:
These make me think we should separate them out to different modules (either now, or Eventually™, e.g. `enrollment::stateful` which can be imported into `nimbus.rs`/`nimbus_client.rs`.

Reply:
It functions fine like this but you're right, it would definitely be better for readability if the stateful pieces are in their own module.

Reply:
Do you think that's worth me making that change in this PR?